### DISCUSSION TITLE
Wave like patterns in Velocity field

### DISCUSSION THREAD

**Discussion by Muthu-V** [firemodels/fds#13638] [Q&A]
Dear FDS developers, I tried to run a simple test case ,  flow over a cube in 2D plane subjected to atmospheric flow to check the flow patterns. After initial transients , I observed some wave like flow structures in the velocity field. The snapshot is attached. Could someone clarify .
Thanks 
![square_2D](https://github.com/user-attachments/assets/89846c6d-7446-4f3f-8a40-177b16c551b4)

 &HEAD CHID='Square_smallDomain'/
&TIME T_END=1000.0/
&DUMP DT_RESTART=100.0, DT_SL3D=0.25, SMOKE3D=.FALSE./
&MISC NOISE=.FALSE./

&SURF ID='sky', FREE_SLIP=.TRUE., COLOR='INVISIBLE' /

&MESH ID='MESH', IJK=500,1,200, XB=3100.0,3600.0,-7.5,7.5,0.0,200.0/
&WIND L=100.0, Z_0=5.0E-3, Z_REF=10.0, SPEED=8.0, DIRECTION=90/

&OBST ID='Obstruction', XB=3362.0,3382.0,-10.0,10.0,0.0,20.0, COLOR='INVISIBLE', SURF_ID='INERT'/ 

&VENT ID='Mesh Vent: MESH [XMAX]', SURF_ID='OPEN', XB=3600.0,3600.0,-7.5,7.5,0.0,200.0/ 
&VENT ID='Mesh Vent: MESH [XMIN]', SURF_ID='OPEN', XB=3100.0,3100.0,-7.5,7.5,0.0,200.0/ 
&VENT ID='Mesh Vent: MESH [ZMAX]', SURF_ID='sky', XB=3100.0,3600.0,-7.5,7.5,200.0,200.0/ 

&SLCF QUANTITY='VELOCITY', VECTOR=.TRUE., PBY=-7.5/
&SLCF QUANTITY='PRESSURE', PBY=-7.5/
&SLCF QUANTITY='U-VELOCITY', VECTOR=.TRUE., PBY=-7.5/
&SLCF QUANTITY='W-VELOCITY', VECTOR=.TRUE., PBY=-7.5/

&TAIL /

**Comment by rmcdermo** [comment]
Well, to me this looks like ringing from the Dirichlet pressure boundary conditions.  This is not uncommon in a low-Mach code.  Try a 3D case with OPEN top bc before you get too stressed about it.  The turbulence cannot develop naturally in 2D.

**Reply by Muthu-V** [reply]
![image](https://github.com/user-attachments/assets/77f9a484-5bf7-4eb8-9933-cbaf1fe01962)

Thank you for your inputs. I set the boundaries to Open and made 3D domain. The waves are not disappearing.

&HEAD CHID='cube_parallel'/
&TIME T_END=1000.0/
&DUMP DT_RESTART=100.0, DT_SL3D=0.25, SMOKE3D=.FALSE./
&MISC NOISE=.FALSE./
&WIND L=100.0, Z_0=5.0E-3, Z_REF=10.0, SPEED=8.0, DIRECTION=90.0/

&MESH ID='MESH-01', IJK=63,100,50, XB=3100.0,3163.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-02', IJK=63,100,50, XB=3100.0,3163.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-03', IJK=63,100,50, XB=3100.0,3163.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-04', IJK=63,100,50, XB=3100.0,3163.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-05', IJK=63,100,50, XB=3100.0,3163.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-06', IJK=63,100,50, XB=3100.0,3163.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-07', IJK=63,100,50, XB=3100.0,3163.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-08', IJK=63,100,50, XB=3100.0,3163.0,0.0,100.0,150.0,200.0/
&MESH ID='MESH-09', IJK=62,100,50, XB=3163.0,3225.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-10', IJK=62,100,50, XB=3163.0,3225.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-11', IJK=62,100,50, XB=3163.0,3225.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-12', IJK=62,100,50, XB=3163.0,3225.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-13', IJK=62,100,50, XB=3163.0,3225.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-14', IJK=62,100,50, XB=3163.0,3225.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-15', IJK=62,100,50, XB=3163.0,3225.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-16', IJK=62,100,50, XB=3163.0,3225.0,0.0,100.0,150.0,200.0/
&MESH ID='MESH-17', IJK=63,100,50, XB=3225.0,3288.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-18', IJK=63,100,50, XB=3225.0,3288.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-19', IJK=63,100,50, XB=3225.0,3288.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-20', IJK=63,100,50, XB=3225.0,3288.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-21', IJK=63,100,50, XB=3225.0,3288.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-22', IJK=63,100,50, XB=3225.0,3288.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-23', IJK=63,100,50, XB=3225.0,3288.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-24', IJK=63,100,50, XB=3225.0,3288.0,0.0,100.0,150.0,200.0/
&MESH ID='MESH-25', IJK=62,100,50, XB=3288.0,3350.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-26', IJK=62,100,50, XB=3288.0,3350.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-27', IJK=62,100,50, XB=3288.0,3350.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-28', IJK=62,100,50, XB=3288.0,3350.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-29', IJK=62,100,50, XB=3288.0,3350.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-30', IJK=62,100,50, XB=3288.0,3350.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-31', IJK=62,100,50, XB=3288.0,3350.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-32', IJK=62,100,50, XB=3288.0,3350.0,0.0,100.0,150.0,200.0/
&MESH ID='MESH-33', IJK=63,100,50, XB=3350.0,3413.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-34', IJK=63,100,50, XB=3350.0,3413.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-35', IJK=63,100,50, XB=3350.0,3413.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-36', IJK=63,100,50, XB=3350.0,3413.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-37', IJK=63,100,50, XB=3350.0,3413.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-38', IJK=63,100,50, XB=3350.0,3413.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-39', IJK=63,100,50, XB=3350.0,3413.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-40', IJK=63,100,50, XB=3350.0,3413.0,0.0,100.0,150.0,200.0/
&MESH ID='MESH-41', IJK=62,100,50, XB=3413.0,3475.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-42', IJK=62,100,50, XB=3413.0,3475.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-43', IJK=62,100,50, XB=3413.0,3475.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-44', IJK=62,100,50, XB=3413.0,3475.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-45', IJK=62,100,50, XB=3413.0,3475.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-46', IJK=62,100,50, XB=3413.0,3475.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-47', IJK=62,100,50, XB=3413.0,3475.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-48', IJK=62,100,50, XB=3413.0,3475.0,0.0,100.0,150.0,200.0/
&MESH ID='MESH-49', IJK=63,100,50, XB=3475.0,3538.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-50', IJK=63,100,50, XB=3475.0,3538.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-51', IJK=63,100,50, XB=3475.0,3538.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-52', IJK=63,100,50, XB=3475.0,3538.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-53', IJK=63,100,50, XB=3475.0,3538.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-54', IJK=63,100,50, XB=3475.0,3538.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-55', IJK=63,100,50, XB=3475.0,3538.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-56', IJK=63,100,50, XB=3475.0,3538.0,0.0,100.0,150.0,200.0/
&MESH ID='MESH-57', IJK=62,100,50, XB=3538.0,3600.0,-100.0,0.0,0.0,50.0/
&MESH ID='MESH-58', IJK=62,100,50, XB=3538.0,3600.0,-100.0,0.0,50.0,100.0/
&MESH ID='MESH-59', IJK=62,100,50, XB=3538.0,3600.0,-100.0,0.0,100.0,150.0/
&MESH ID='MESH-60', IJK=62,100,50, XB=3538.0,3600.0,-100.0,0.0,150.0,200.0/
&MESH ID='MESH-61', IJK=62,100,50, XB=3538.0,3600.0,0.0,100.0,0.0,50.0/
&MESH ID='MESH-62', IJK=62,100,50, XB=3538.0,3600.0,0.0,100.0,50.0,100.0/
&MESH ID='MESH-63', IJK=62,100,50, XB=3538.0,3600.0,0.0,100.0,100.0,150.0/
&MESH ID='MESH-64', IJK=62,100,50, XB=3538.0,3600.0,0.0,100.0,150.0,200.0/


&OBST ID='Obstruction', XB=3362.0,3382.0,-10.0,10.0,0.0,20.0, SURF_ID='INERT'/ 

&VENT ID='Mesh Vent: MESH [XMAX]', SURF_ID='OPEN', XB=3600.0,3600.0,-100.0,100.0,0.0,200.0/ 
&VENT ID='Mesh Vent: MESH [XMIN]', SURF_ID='OPEN', XB=3100.0,3100.0,-100.0,100.0,0.0,200.0/ 
&VENT ID='Mesh Vent: MESH [YMAX]', SURF_ID='OPEN', XB=3100.0,3600.0,100.0,100.0,0.0,200.0/ 
&VENT ID='Mesh Vent: MESH [YMIN]', SURF_ID='OPEN', XB=3100.0,3600.0,-100.0,-100.0,0.0,200.0/ 
&VENT ID='Mesh Vent: MESH [ZMAX]', SURF_ID='OPEN', XB=3100.0,3600.0,-100.0,100.0,200.0,200.0/ 

&SLCF QUANTITY='VELOCITY', VECTOR=.TRUE., ID='Y = 0.0 m: Velocity', PBY=0.0/
&SLCF QUANTITY='PRESSURE', ID='Y = 0.0 m: Pressure', PBY=0.0/
&SLCF QUANTITY='U-VELOCITY', VECTOR=.TRUE., ID='Y = 0.0 m: U-Velocity', PBY=0.0/
&SLCF QUANTITY='W-VELOCITY', VECTOR=.TRUE., ID='Y = 0.0 m: W-Velocity', PBY=0.0/


&TAIL /

**Reply by rmcdermo** [reply]
Try adding a coarse mesh outside the current domain to extend the boundary.  What happens in low-Mach codes is that the pressure boundaries are reflective and you can get a numerical ringing that mimics a resonance in a chamber.  If you change the dimensions of the chamber you change the resonance, so moving the boundaries may help.  In the end, though, depending on what you are trying to model, this noise might not be a major issue.

**Reply by Muthu-V** [reply]
Thank you for your insights. I see those wave patters are appearing even inside the domain. 
![image](https://github.com/user-attachments/assets/36f6d71f-da90-4d77-8fd3-a819db3bbdbd)

**Reply by mcgratta** [reply]
These patterns result from approximating the partial derivatives in the momentum equation with second-order accurate finite differences both in space and time.  They represent, to some extent, the numerical error in making these approximations.

**Comment by marcosvanella** [comment]
Dispersion errors from centered finite difference formulas used on the momentum equations (in particular advection terms). The upside is, doing things this way you conserve conserved quantities, in particular kinetic energy.
________________________________

**Reply by Muthu-V** [reply]
Thank you for your insights, Kevin and Marcos. However the field are smooth after averaging, just like the second-order upwind based RANS schemes :-)
![image](https://github.com/user-attachments/assets/8cf46983-c137-4319-a4c1-3dbaaf02ecb2)
